include/mbgl/util/thread.hpp

```diff
@@ -40,24 +40,33 @@ template<class Object>
 class Thread : public Scheduler {
 public:
     template <class... Args>
-    Thread(const std::string& name, Args&&... args) {
-        std::promise<void> running;
-
-        thread = std::thread([&] {
+    Thread(const std::string& name, Args&&... args)
+        : object(std::make_unique<Actor<Object>>()) {
+
+        std::unique_ptr<std::promise<void>> running_ = std::make_unique<std::promise<void>>();
+        running = running_->get_future();
+        
+        auto capturedArgs = Actor<Object>::captureArguments(std::forward<Args>(args)...);
+
+        thread = std::thread([
+            this,
+            name,
+            capturedArgs,
+            runningPromise = std::move(running_)
+        ] {
             platform::setCurrentThreadName(name);
             platform::makeThreadLowPriority();
 
             util::RunLoop loop_(util::RunLoop::Type::New);
             loop = &loop_;
 
-            object = std::make_unique<Actor<Object>>(*this, std::forward<Args>(args)...);
-            running.set_value();
+            object->activate(*this, std::move(capturedArgs));
```

Comment:
Pass `loop_` instead of `*this`, and then `Thread` no longer needs to inherit from `Scheduler` or define `schedule`.